### PR TITLE
Deprecate Deployment .spec.rollbackTo field 

### PR DESCRIPTION
**Pull request by janetkuo**
~Depends on #48746~ (merged)
xref: #46934, #49135

1. Deprecate Deployment field `.spec.rollbackTo` in `extensions/v1beta1` and `apps/v1beta1`, and remove the same field and `/rollback` endpoint from `apps/v1beta2` Deployment. 
1. Add an annotation `deprecated.deployment.rollback.to` in `apps/v1beta2` for conversion to/from other versions. 

Note: `apps/v1beta2` is new in 1.8 (and WIP), so it is okay to make breaking changes to it. 

```release-note
Deprecate Deployment .spec.rollbackTo field 
```

### CODE REVIEW

**Comment by janetkuo** [comment]
@kubernetes/sig-apps-api-reviews

**Comment by janetkuo** [comment]
I need to update conversion_test.go as well

**Comment by janetkuo** [comment]
Seems a test-infra issue
```
Something went wrong: failed to acquire k8s binaries: error during /go/src/k8s.io/release/push-build.sh --nomock --verbose --bucket=kubernetes-release-pull --ci --gcs-suffix=/pull-kubernetes-e2e-kops-aws: exit status 2
```

/test pull-kubernetes-e2e-kops-aws

**Comment by janetkuo** [comment]
The test failure is #49672

**Comment by kargakis** [comment]
/approve no-issue

cc: @mfojtik @tnozicka

**Comment by janetkuo** [comment]
I have added some more commits to enable and use apps/v1beta2 deployment's auto generated conversions, see more in #49738

**Comment by kargakis** [comment]
@smarterclayton @tnozicka @mfojtik any thoughts here?

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 const (
 	ControllerRevisionHashLabelKey = "controller-revision-hash"
 	StatefulSetRevisionLabel       = ControllerRevisionHashLabelKey
+	DeprecatedRollbackTo           = "deprecated.deployment.rollback.to"
```
Comment (kow3ns) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: Is this the right naming convention?

**Review by kow3ns — APPROVED** [review]

**Comment by erictune** [comment]
/approve

approved for API change

**Comment by janetkuo** [comment]
Just another rebase

**Comment by bgrant0607** [comment]
This is fine. It could be implemented in an external controller using annotations, and consistency across the workload controllers is more important.

/approve

**Pull request merged**